src/shell.rs

```diff
@@ -1,22 +1,25 @@
 use std::path::PathBuf;
+use std::rc::Rc;
 
-pub struct Shell<'a> {
+use config::Config;
+
+pub struct Shell<T: Config> {
     pub name: String,
-    pub root_path: &'a PathBuf,
+    pub config: Rc<T>,
 }
 
-impl<'a> Shell<'a> {
```

Comment:
AND I got rid of this lifetime parameter!  Once again, my experience continues to tell me that if you're using lifetime parameters, you're doing something kinda janky.